// @angular-magic/ngx-gp-autocomplete/esm2022/lib/ngx-gp-autocomplete.service.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class NgxGpAutocompleteService {
    constructor() {
        this.defaultOptions = {};
    }
    setOptions(options) {
        this.defaultOptions = { ...this.defaultOptions, ...options };
    }
    getOptions() {
        return this.defaultOptions;
    }
    enableGooglePersistenceCheck() {
        this.verifyGooglePersistence = true;
    }
    getGooglePersistenceCheck() {
        return this.verifyGooglePersistence;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.2.6", ngImport: i0, type: NgxGpAutocompleteService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.2.6", ngImport: i0, type: NgxGpAutocompleteService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.2.6", ngImport: i0, type: NgxGpAutocompleteService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
